_posts/2023-10-02-intellij-quarkus-recap.adoc

```diff
@@ -0,0 +1,178 @@
+Properties completion in `application.properties` files now displays icons and documentation (provided `Preferences | Editor | General | Code Completion | Show the documentation popup` is enabled):
+image:properties-completion.png[Quarkus properties completion]
+
+Completion is also available for enum values:
+image:properties-completion-enum.png[Quarkus properties enum completion]
+
+=== Inlay hints
+Properties using expression values are now resolved as inlay hints:
+image:properties-inlay-hint.png[Resolved property expression as inlay hint]
+
+Inlay hints can be disabled in `Preferences | Languages & Frameworks | MicroProfile | Properties`
+
+== Java editing improvements
+
+=== New codelens for Vert.x reactive routes
+Vert.x reactive routes now show a clickable codelens, that will open the browser to the configured URL, similar to JAX-RS endpoints:
+
+image:reactive-routes.png[Reactive routes]
+
+== Qute improvements
+We've spent significant effort on making the best Qute support around.
+
+=== Completion redux
+Editing Qute templates is now better than ever, with new brackets autoclose, improved HTML integration, new completion icons. In particular, the enhanced snippet completion, that behaves like IntelliJ's live templates, available for different Qute sections:
+
+image:qute-completion-snippets.gif[Qute snippet completion]
+
+=== Improved {#insert} and {#include} support
+The `{#insert}` and `{#include}` sections can be used to specify the extended template to include in the current template. This release extends the support for these sections with validation, document link, completion, and code lens.
+
+=== Navigation between templates/code
+You can navigate by `Ctrl+click` (`Cmd+click` on Mac), from one template to another (referenced as user tags, via `{#include}` or `{#insert}`), or from a template definition from a java file to the actual file:
+image:qute-template-navigation.gif[Qute navigation]
+
+=== New template fragment support
+Fragment sections define a part of the template that can be treated and rendered as a separate template. Support for fragments includes snippet completion, hover and validation.
+In a java class, you can navigate to a referenced `#fragment` via code lens and document link
+image:qute-fragment-support.gif[]
+
+=== Renarde support
+
+Quarkus Tools now provides support for the https://docs.quarkiverse.io/quarkus-renarde/dev/index.html[Renarde] web framework:
+
+- Public methods of sub-classes of a Renarde `Controller` automatically show a clickable codelens, even though they're lacking a HTTP method annotation, as they're treated as valid JAX-RS endpoints:
+image:qute-renarde-codelens.png[Renarde endpoint codelens]
+
+- Completion on the Renarde `uri` and `uriabs` namespaces lists available Renarde controllers:
+image:qute-renarde-completion.png[Renarde uri completion]
+
+- Section `{#}` completion displays available https://docs.quarkiverse.io/quarkus-renarde/1.x/index.html#_renarde_tags[Renarde tags] 
```

Comment:
This collect of user tags is generic and not hard coded (it scans each templates.tags folder from JAR).It means that:

 * as it is not hard coded, the available renarde user tags could be different from different renarde version
 * it will work for any library which hosts user tags. So if you have Quarkus Web Bundler in your classpath you will have those user tags https://github.com/quarkiverse/quarkus-web-bundler/tree/main/runtime/src/main/resources/templates/tags
 *

Reply:
OK, moved to the Qute completion section